### ISSUE TITLE
Setting up guide in README outdated

### ISSUE THREAD

**Issue by RasmusKlett** [lambdacms/lambdacms#14]
I just set up a fresh install following the README guide step by step. I had two problems, at least one of which seems to be fixed in the setup script you provide. 

1:
While executing the patches, I got one error:

> $ for f in $TMP_CLONE/yesod-scaffold-patches/*.patch; do patch -p1 < $f; done
> patching file Application.hs
> Hunk #3 succeeded at 77 with fuzz 2.
> patching file config/models
> patching file config/routes
> patching file config/settings.yml
> Hunk #1 succeeded at 27 (offset 4 lines).
> patching file Foundation.hs
> patching file Handler/Home.hs
> patching file Model.hs
> patching file project_name.cabal
> Hunk #2 succeeded at 82 with fuzz 2.
> patching file Roles.hs
> patching file Settings.hs
> patching file templates/homepage.hamlet
> patching file templates/homepage.julius
> patching file templates/homepage.lucius
> patching file test/Handler/HomeSpec.hs
> patching file test/TestImport.hs
> Hunk #2 FAILED at 48.
> 1 out of 2 hunks FAILED -- saving rejects to file test/TestImport.hs.rej

`
The contents of TestImport.hs.rej afterwards is:

> --- test/TestImport.hs    2015-07-22 22:52:20.271915262 +0200
> +++ test/TestImport.hs    2015-07-22 22:52:23.023910152 +0200
> @@ -48,8 +47,8 @@
> 
> ```
>  -- Aside: SQLite by default *does not enable foreign key checks*
>  -- (disabling foreign keys is only necessary for those who specifically enable them).
> ```
> -    let settings = appSettings app  
> -    sqliteConn <- rawConnection (sqlDatabase $ appDatabaseConf settings)  
> -    let settings = appSettings app
> -    sqliteConn <- rawConnection (sqlDatabase $ appDatabaseConf settings)
>    disableForeignKeys sqliteConn
>   
>    let logFunc = messageLoggerSource app (appLogger app)

It seems this patch is assumig a sqlite database, and I am going with Postgres. So I just ignored this particular patch, and did the rest of the patches. This seems to have worked, but it was very disruptive to the flow of the guide.

2:
After the patching was done, I seemed to be missing a lot of dependencies, most importantly lambdacms-core itself. Poking through the setup script, I discovered it adds the following to stack.yaml extra-deps:

> extra-deps:
> - lambdacms-core-0.3.0.2
> - friendly-time-0.4
> - lists-0.4.2
> - list-extras-0.4.1.4

After adding these to my own stack.yaml, I was able to finish the build. I am guessing this should be stated in the guide?

**Comment by cies** [comment]
Yes you are right. I'm actually working on gettting the scaffolds to work with `stack new`; I ran into some problems and then also ran out of time... Sorry for that, and sorry for my late reply.

I will gladly accept a PR on the documentation and/or on the script itself. But if I find time I will first move the whole thing to `stack new` (this needs more then a simple PR, as [it needs some infrastructure to stay in check with future versions of Yesod](https://github.com/lambdacms/lambdacms/issues/4)).

Thanks for your interest in LambdaCms, I hope it all worked out for you! If you have any questions feel free to ask them here.

**Comment by RasmusKlett** [comment]
I think all the problems would be solved by a stack scaffold, so it might not be worth updating soon-to-be-outdated documentation ;)
LambdaCms seems very interesting, I am still considering it for some projects :) It seems to be just what I need, I'm just unsure of its maturity.

**Comment by cies** [comment]
It's safe to say it's not mature. But the good thing is that it is not
trying to do too much, and it pretty extendable. So it is just a thin layer
on top of Yesod (which I consider mature), that allows us to create and
share admin-sections in a uniform way.

**Comment by RasmusKlett** [comment]
This is becoming quite unrelated, feel free to reply by message instead.
I get that the scope is reasonably small, but coming from the Django admin, does it offer enough hooks for customisation? I haven't dug into it yet, but plan on doing so. 
Also, what is the future of the project? Do you consider it feature-complete, or is there a wishlist of future improvements? If we end up using it for the semi-commercial application we're brewing, we would probably end up contributing to the project, but we don't want to bet on something with no future plans.. There just doesn't seem to be a very supported option in the Haskell ecosystem yet :P

**Comment by cies** [comment]
> does it offer enough hooks for customisation?

Which hooks in particular are you looking for?

> I haven't dug into it yet, but plan on doing so.

That's good. There is very little code to it, so a simple "read all" should
give you a good idea of where we are.

> Also, what is the future of the project?

This is not a mature project like Django. You will have to bring some to
make use of it. Most importantly, this project is project is mostly giving
an infrastructure to build websites with admin backend on top of Yesod; in
a fashion that promotes code re-use.

It follows the model of Rails and RefineryCMS: RefCMS is a small layer on
top of Rails, just 20k lines of code, by which functionality can be
packaged (gems) in subsites (Rails engines) that can be re-used across
RefCMS installations.

> Do you consider it feature-complete, or is there a wishlist of future
> improvements?

Not feature complete. But not far from there either.

The most important feature I have in mind is to flesh out a way to expose
admin backend "widgets" that can be re-used by other admin backend modules.
For example an image picker widget that is exposed by the media library
package; you can use this widget in your hypothetical admin backend of you
"products" package.

> If we end up using it for the semi-commercial application we're brewing,
> we would probably end up contributing to the project, but we don't want to
> bet on something with no future plans.. There just doesn't seem to be a
> very supported option in the Haskell ecosystem yet :P

Nope. I really think this is your best bet. But be sure to look around.
There's HsCMS and ClckWrks, compared to them I'd pick LambdaCms.

The bottom line is that, since LambdaCms is a rather thin layer over Yesod,
you end up with a Yesod app. You can always take it from there, therefor
you are "buying in" to Yesod more then LambdaCms. And Yesod is, IMHO, the
defacto standard webframework for Haskell at the moment.

All the best with your choice, and do not hesitate to ask any questions you
may have.

**Comment by RasmusKlett** [comment]
Thanks for all your time, you are mostly confirming my impression of it, which is good ;)
We are already on Yesod, which of course makes this tempting. I will test it out and make a decision within the coming weeks.

**Comment by cies** [comment]
Please let us know what you found!

**Comment by RasmusKlett** [comment]
Will do

2016-03-07 15:22 GMT+01:00 Cies Breijs notifications@github.com:

**Comment by RasmusKlett** [comment]
I think I've unfortunately come to the conclusion that this doesn't suit my needs. I was looking for something with a slightly larger scope. I was hoping this provided a way to wire a model up so LambdaCMS could provide some default templates and methods for CRUD operations, but it seems extensions have to implement all of that by themselves. 
I guess that's fine, but to us the project's functionality doesn't justify the cost of integrating and depending on it being maintained.

**Comment by cies** [comment]
Thanks for coming back to say this. I understand what you are looking for (not sure if I know such a project in any language, probably Administrator by Thoughtbot comes closest).

All the best for your project!

**Comment by cies** [comment]
Closed the issue mistakenly, as it is still valid.

**Comment by RasmusKlett** [comment]
Well, I have primarily worked in Django, so maybe I'm just spoiled :P Their system has it, and it is such an unbelievable timesaver. But I realize now that it's probably pretty rare.

**Comment by dfordivam** [comment]
Hi Cies

I am trying to use 'stack new' approach (as it the only method now), but got lots of failures in patches. And just by looking at the patch files I get no clue on how to merge. Are you able to get time and fix this issue? or can you post a patched project which works correctly.

**Comment by cies** [comment]
You could have a look at the project for the lambdacms.org website:

https://github.com/lambdacms/lambdacms.org
